dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java

```diff
@@ -391,6 +254,174 @@ private void updateAuditStatus() throws DotPublisherException, DotDataException
+                            }
+                        } catch (Exception e) {
+                            // An error occurred when retrieving the end-point's audit info.
+                            // Usually caused by a network problem.
+                            Logger.error(PublisherQueueJob.class,
+                                    String.format(
+                                            "An error occurred when accessing end-point '%s' with IP %s: %s",
+                                            targetEndpoint.getServerName(), targetEndpoint.getAddress(),
+                                            e.getMessage()),
+                                    e);
+                            String failedAuditUpdate = "failed-remote-group-" + System.currentTimeMillis();
+                            EndpointDetail detail = new EndpointDetail();
+                            detail.setStatus(Status.FAILED_TO_PUBLISH.getCode());
+                            endpointTrackingMap.put(failedAuditUpdate, map(failedAuditUpdate, detail));
+
+                            PushPublishLogger.log(this.getClass(), "Status update: Failed to publish bundle");
+                        }
+                    } else {
+                        PublishAuditStatus pas = pubAuditAPI.getPublishAuditStatus(bundleAudit.getBundleId());
+                        endpointTrackingMap.putAll(pas.getStatusPojo().getEndpointsMap());
+                    }
+                }
+            }
+        }
+
+        return endpointTrackingMap;
+	}
+
+	@NotNull
+	private void updateBundleStatus(PublishAuditStatus bundleAudit,
+									Map<String, Map<String, EndpointDetail>> endpointTrackingMap,
+									GroupPushStats groupPushStats) throws DotDataException, DotPublisherException {
+		Status bundleStatus;
+		PublishAuditHistory localHistory = bundleAudit.getStatusPojo();
+
+		if ( localHistory.getNumTries() >= MAX_NUM_TRIES && (groupPushStats.countGroupFailed() > 0
+                || groupPushStats.countGroupPublishing() > 0) ) {
+            // If bundle cannot be installed after [MAX_NUM_TRIES] tries
+            // and some groups could not be published
+            List<Environment> environments = APILocator.getEnvironmentAPI().findEnvironmentsByBundleId(bundleAudit.getBundleId());
+            for ( Environment environment : environments ) {
+                APILocator.getPushedAssetsAPI().deletePushedAssets(bundleAudit.getBundleId(), environment.getId());
+            }
+            PushPublishLogger.log(this.getClass(), "Status Update: Failed to publish");
+            bundleStatus = Status.FAILED_TO_PUBLISH;
+            pubAuditAPI.updatePublishAuditStatus(bundleAudit.getBundleId(), bundleStatus, localHistory);
+            pubAPI.deleteElementsFromPublishQueueTable(bundleAudit.getBundleId());
+        } else if (groupPushStats.countGroupFailed() > 0 && (groupPushStats.countGroupOk() + groupPushStats.countGroupFailed()) == endpointTrackingMap.size()) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 160 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)